tests/vm_test.go

```diff
@@ -1496,6 +1496,82 @@ status:
 			waitForResourceDeletion(k8sClient, "pods", expectedPodName)
 		})
 
+		Context("Deleting a running VM with high TerminationGracePeriod via command line with", func() {
+			It("--force --grace-period=0 should force delete the VM", func() {
+				By("getting a VM with a high TerminationGracePeriod")
+				vmi := libvmi.New(
+					libvmi.WithResourceMemory("128Mi"),
+					libvmi.WithTerminationGracePeriod(1600),
+				)
+				vm := tests.NewRandomVirtualMachine(vmi, true)
+				vm.Namespace = testsuite.GetTestNamespace(vm)
+
+				vmJson, err := tests.GenerateVMJson(vm, workDir)
+				Expect(err).ToNot(HaveOccurred(), "Cannot generate VMs manifest")
+
+				By("Creating VM using k8s client binary")
+				_, _, err = clientcmd.RunCommand(k8sClient, "create", "-f", vmJson)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Waiting for VMI to start")
+				Eventually(ThisVMIWith(vm.Namespace, vm.Name), 240*time.Second, 1*time.Second).Should(BeRunning())
+
+				By("Checking that VM is running")
+				stdout, _, err := clientcmd.RunCommand(k8sClient, "describe", "vmis", vm.GetName())
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(vmRunningRe.FindString(stdout)).ToNot(Equal(""), "VMI is not Running")
+
+				By("Sending a force delete VM request using k8s client binary")
+				_, _, err = clientcmd.RunCommand(k8sClient, "delete", "vm", vm.GetName(), "--grace-period=0", "--force", "--wait=false")
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Verifying the VM gets deleted")
+				waitForResourceDeletion(k8sClient, "vms", vm.GetName())
+
+				By("Verifying pod gets deleted")
+				expectedPodName := getExpectedPodName(vm)
+				waitForResourceDeletion(k8sClient, "pods", expectedPodName)
+			})
```

Comment:
I think that both `It` can be grouped with something like this:
```
Context("Deleting a running VM with high TerminationGracePeriod via command line", func() {
			DescribeTable("should force delete the VM", func(deletionArgs []string) {
			    [...]
				By("Sending a force delete VM request using k8s client binary")
				args := append([]string{"delete", "vm", vm.GetName()}, deletionArgs...)
				_, _, err = clientcmd.RunCommand(k8sClient, args...)
				Expect(err).ToNot(HaveOccurred())
				[...]
			},
				Entry("when --force and --grace-period=0 are provided", []string{"--force", "--grace-period=0", "--wait=false"}),
				Entry("when --now are provided", []string{"--now"}),
			)
})
```

Reply:
Updated